### PR TITLE
Eventing StackDriver internal metrics

### PR DESCRIPTION
**Pull request by nachocano**
- Making eventing metrics internal for stackdriver as we will tag them with container and pod names to avoid problems when reporting the same metric from different pods/containers. Doing so for KnativeBroker and KnativeTrigger only.
- Added some constants.
- Note that this shouldn't be here. There are TODOs in the code with issue numbers to do a proper refactor and move eventing and serving specific metrics out of pkg.
- Removing eventSource and filterSource tags from broker/trigger stuff to keep the cardinality manageable. Leaving it for KnativeSource for now.

- We should cherry-pick this for 0.10.

### CODE REVIEW

File: metrics/metricskey/constants_eventing.go [hunk 1]
```diff
@@ -59,16 +59,16 @@ var (
 		LabelLocation,
 		LabelClusterName,
 		LabelNamespaceName,
-		LabelName,
```
Comment (yanweiguo) [inline metrics/metricskey/constants_eventing.go]: Is `LabelName` still used somewhere? Could we remove its definition?

File: metrics/stackdriver_exporter_test.go [hunk 2]
```diff
@@ -185,15 +185,11 @@ func TestGetMonitoredResourceFunc_UseKnativeBroker(t *testing.T) {
 		if got != testEventType {
 			t.Errorf("expected new tag: %v, got: %v", eventTypeKey, newTags)
 		}
-		got = getResourceLabelValue(metricskey.LabelEventSource, newTags)
```
Comment (yanweiguo) [inline metrics/stackdriver_exporter_test.go]: Do you want to check `LabelSourceName` instead?

File: metrics/metricskey/constants_eventing.go (hunk 1 above)
Comment (nachocano) [reply metrics/metricskey/constants_eventing.go]: I'll rollback the change on sources. If I do so, then nothing should break in other repos, and will make it easy for cherry-picking to 0.10.
So, yeah, this Name will be used.

File: metrics/stackdriver_exporter_test.go (hunk 2 above)
Comment (nachocano) [reply metrics/stackdriver_exporter_test.go]: Yeah, but I'll change it to Name.

Thanks for taking a look! Will ping you tmrw for another round :)

Comment (nachocano) [reply metrics/stackdriver_exporter_test.go]: this label has been dropped from KnativeBroker, so it's fine not to check this

**Pull request merged**